statik/autogen.py

```diff
@@ -0,0 +1,92 @@
+import os.path
+
+import yaml
+
+from statik.config import *
+from statik.fields import *
+from statik.models import *
+from statik.project import *
+from statik.utils import *
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = ['autogen']
+
+def autogen(project_path):
+    """Autogenerates views and templates for all the models in the project."""
+    generate_quickstart(project_path)
+
+    project = StatikProject(project_path)
+    project.config = StatikConfig(project.config_file_path)
+
+    models = project.load_models()
+
+    logger.info('Creating view and template for home page (index.html).')
+    generate_yaml_file(os.path.join(project_path, StatikProject.VIEWS_DIR, 'index.yaml'),
+                        {   
+                            'path': '/',
+                            'template': 'index'
+                        }
+    )
+    generate_index_file(os.path.join(project_path, StatikProject.TEMPLATES_DIR, 'index.jinja2'))
+
+    for (modelname, model) in models.items():
+        logger.info('Creating view and template for model: %s' % model.name)
+        generate_yaml_file(os.path.join(project_path, StatikProject.VIEWS_DIR, '%s.yaml' % model.name),
+                            {   
+                                'path': {
+                                    'template': '/%s/{{ %s.pk }}' % (model.name, model.name),
+                                    'for-each': {
+                                        '%s' % model.name: 'session.query(%s).all()' % model.name
+                                    }
+                                },
+                                'template': ('%s' % model.name),
+                            }
+        )
+        generate_model_file(os.path.join(project_path, StatikProject.TEMPLATES_DIR, '%s.jinja2' % model.name),
+                            model.name,
+                            model.fields.values())
+
+
+def generate_yaml_file(filename, contents):
+    """Creates a yaml file with the given content."""
+    with open(filename, 'w') as file:
+        file.write(yaml.dump(contents, default_flow_style=False))
+
+
+def generate_index_file(filename):
+    """Constructs a default home page for the project."""
+    with open(filename, 'w') as file:                            
+        file.write("<!DOCTYPE html>" + os.linesep)
+        file.write("<html>" + os.linesep)
+        file.write("    <head><title>{{ site_title }}</title></head>" + os.linesep)
+        file.write("    <body>" + os.linesep)
+        file.write("        index" + os.linesep)
+        file.write("    </body>" + os.linesep)
+        file.write("</html>" + os.linesep)
```

Comment:
Keep things consistent. The project (mostly) uses single quotes, I think it would be a good idea to use those here too. (unless there is a good reason not to, but as far as i'm concerned I don't know of it)

Reply:
store the large string in a file, as a jinja template.

run the jinja engine on the template, providing `site_title` as a variable.

(this repo has lots of jinja2 templating logic which you can re-use)